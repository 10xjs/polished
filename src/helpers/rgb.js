// @flow

/**
 * Returns a string value for the color. The returned result is the smalles possible hex notation.
 *
 * @example
 * // Styles as object usage
 * const styles = {
 *   background: rgb('255, 205, 100'),
 *   background: rgb(255, 205, 100),
 *   background: rgb({ red: 255, green: 205, blue: 100 }),
 * }
 *
 * // styled-components usage
 * const div = styled.div`
 *   background: ${rgb('255, 205, 100')};
 *   background: ${rgb(255, 205, 100)};
 *   background: ${rgb({ red: 255, green: 205, blue: 100 })};
 * `
 *
 * // CSS in JS Output
 *
 * element {
 *   background: TODO
 * }
 */

type RgbColor = {
  red: number,
  green: number,
  blue: number,
}

function rgb(value: string | RgbColor | number, green?: number, blue?: number): string {
  return '#FFF'
}

export default rgb
