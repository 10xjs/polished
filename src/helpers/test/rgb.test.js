// @flow
import rgb from '../rgb'

describe('em', () => {
  it('should convert a rgb string to a hex color', () => {
    expect({ background: rgb('255, 205, 100') }).toMatchSnapshot()
  })
})
