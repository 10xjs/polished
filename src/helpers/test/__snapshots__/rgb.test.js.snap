exports[`em should convert a rgb string to a hex color 1`] = `
Object {
  "background": "#FFCD64",
}
`;
